smart-contracts/for-eth-devs/gas-model.md

```diff
@@ -1 +1,85 @@
-TODO
\ No newline at end of file
+---
+title: Gas Model on the Polkadot Hub
+description: Learn how gas estimation, pricing, and weight mapping work in the Polkadot Hub.
+---
+
+# Gas Model
+
+## Overview
+
+The Polkadot Hub implements a gas model that bridges Ethereum's familiar gas concept with Polkadot's more sophisticated resource metering system. This page explains how gas works in the Polkadot Hub and what developers need to know when building smart contracts.
+
+## Understanding Resources in the Polkadot Hub
+
+Unlike Ethereum, which uses a single gas value to measure everything, the Polkadot Hub tracks three separate resources:
+
+- **`ref_time`** measures computational time - this is the closest equivalent to traditional gas and represents how much CPU time your contract execution consumes.
+- **`proof_size`** measures the amount of state data that validators need to verify. When your contract reads from storage or makes state queries, this metric tracks the size of the proofs needed to validate those operations.
+- **`storage_deposit`** is a native balance that gets temporarily locked when your contract creates new storage entries. This prevents state bloat by requiring contracts to pay for the storage they use. The deposit is returned when the storage is freed.
+
+For Ethereum wallet compatibility, the Polkadot Hub's RPC layer automatically maps these three dimensions into a single gas value that wallets can understand and display to users.
+
+## Gas vs Weight
+
+When you interact with the Polkadot Hub through an Ethereum wallet, you see familiar gas values. Under the hood, the runtime works with "weight" - a two-dimensional metric that combines `ref_time` and `proof_size`.
+
+The system continuously translates between these representations: converting weight to gas when estimating costs, and converting gas back to weight when executing transactions.
+
+## How Gas Estimation Works
+
+When your wallet requests a gas estimate (`eth_estimateGas`), the Polkadot Hub performs a dry-run of your transaction. This test execution discovers:
+
+- How much computational time the contract will consume (`ref_time`)
+- How much state data needs to be verified (`proof_size`)
+- Whether any storage deposits are required (`storage_deposit`)
```

Comment:
I see later in this document a reference to length fee and multiplier fee. Is this taken into account in the `eth_estimatedGas`?

Reply:
Okay can you add this to the page so its super clear?